libpromises/generic_agent.c

```diff
@@ -1968,3 +1968,100 @@ void SetupSignalsForAgent(void)
     signal(SIGUSR1, HandleSignalsForAgent);
     signal(SIGUSR2, HandleSignalsForAgent);
 }
+
+void ShowContextsFormatted(EvalContext *ctx)
```

Comment:
OK; moved to generic_agent.c because I do think that feel like the right place for it. If not, tell me where, or move it after merge, it's a trivial change.